File: unidade-II/pratica-arquivos-2/questao3/questao3.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct fruta
{
    char nome[20];
    float preco;

} Fruta;


int main()
{
    FILE* fl = fopen("frutas.txt", "a"); // Abre o arquivo com "a" para inserir as novas frutas ao final do arquivo
    if (fl == NULL)
    {
        printf("Erro ao abrir o arquivo!\n");
        exit(1);
    }

    Fruta fruta;
    
    char continuar[2];
    do
    {
        printf("Digite o nome da fruta: ");
        scanf(" %[^\n]", fruta.nome);
        printf("Digite o preço da fruta: ");
        scanf("%f", &fruta.preco);


        fprintf(fl, "%s\tR$ %2.f\n", fruta.nome, fruta.preco);

        printf("Deseja continuar? (S/n) ");
        scanf(" %c", continuar);

    } while (continuar[0] == 'S' || continuar[0] == 's');
    
    

    fclose(fl);

    return 0;
}

File: unidade-II/pratica-arquivos-2/questao3/questao3.py

class Fruta:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

def main():
    while True:
        nome = input("Digite o nome da fruta (ou 'sair' para encerrar a execução): ")

        if nome.lower() == 'sair':
            break

        preco = float(input("Digite o preço: "))

        fruta = Fruta(nome, preco)

        with open("frutas_py.txt", "a") as arquivo:
            arquivo.write(f'{fruta.nome} R$ {fruta.preco:.2f}\n')
    
    print("Informações registradas na no arquivo 'frutas_py.txt'")

if __name__ == "__main__":
    main()